File: AdjacencyMatrix.py
import random
import numpy  # type: ignore


def adjacencyMatrix():
    print("Type how much nodes are in your graph :")
    nodes = int(input())
    print("How many arcs :")
    arcs = int(input())
    counter = 0
    matrix = numpy.zeros((nodes,nodes))
    deg = numpy.zeros((nodes,2))

    
    # adding arcs 
    while(counter < arcs):
        i = random.randint(0,nodes-1)
        j = random.randint(0,nodes-1)
        if matrix[i][j] == 0 :
            matrix[i][j] = 1
            counter+=1

    # caluculate deg
    for i in range(nodes):
        degP = 0
        degM = 0
        for j in range(nodes):
            if matrix[i][j] == 1:
                degP+=1
            if matrix[j][i] == 1:
                degM+=1
        deg[i][0] = degP
        deg[i][1] = degM

    # Calculer le deg pour assurer le fait que le graph est oriente      
    sommeDP = 0
    sommeDM = 0
    for i in range(nodes):
        sommeDP+=deg[i][0]
    for i in range(nodes):
        sommeDM+=deg[i][1]
    
    
    # Display matrix
    for r in matrix:
        print(r)

    # Display the deg matrix
    for r in deg:
        print(r)
    print(f"la somme des degres+ vaut : {int(sommeDP)} et les degres- vaut {int(sommeDM)}")

    # Interpretation du graph
    for i in range(nodes):
        print("----------------------------------------")
        for j in range(nodes):
            if matrix[i][j] == 1:
                print(f"the node {i} is connected to the node {j}")

adjacencyMatrix()



    

File: AdjacencyMatrix.java
import java.util.Random;
import java.util.Scanner;

public class AdjacencyMatrix {
    public static void calculerDeg(int n, int[][] matrice) {

        int[][] deg = new int[n][2];
        int sommeP = 0, sommeM = 0;
        for (int i = 0; i < n; i++) {
            int degP = 0, degM = 0;
            for (int j = 0; j < n; j++) {
                if (matrice[i][j] == 1) {
                    degP++;
                }
                if (matrice[j][i] == 1) {
                    degM++;
                }
            }
            deg[i][0] = degP;
            deg[i][1] = degM;
        }
        System.out.println("-------Matrice Deg-------");
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < 2; j++) {
                System.out.print(" " + deg[i][j] + " ");
            }
            System.out.println();
        }
        for (int i = 0; i < n; i++) {
            sommeP += deg[i][0];
            sommeM += deg[i][1];
        }
        System.out.println(
                "Somme de Deg+ = " + sommeP + " et Deg- = " + sommeM + "\n--------------------------------------");
    }

    public static void main(String[] args) {

        Random random = new Random();
        Scanner scan = new Scanner(System.in);
        int n, arc, i, j, counter = 0;

        System.out.print("Type the number of nodes :");
        n = scan.nextInt();
        // forçage des arcs
        while (true) {
            System.out.print("Type the number of arcs :");
            arc = scan.nextInt();
            if (arc < Math.pow(n, 2)) {
                break;
            }
        }

        // pour la matrice
        int[][] matrice = new int[n][n];
        // pour calculer les deg+ et deg- afin d'assurer que le graphe est oriente

        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++) {
                matrice[i][j] = 0;
            }
        }

        // pour assurer le nb des arcs via un compteur qui compte le nombre des 1 dans
        // la matrice
        while (counter < arc) {
            i = random.nextInt(n);
            j = random.nextInt(n);
            if (matrice[i][j] == 0) {
                matrice[i][j] = 1;
                counter++;
            }
        }

        // afficher la matrice
        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++) {
                System.out.print(" " + matrice[i][j] + " ");
            }
            System.out.println();
        }

        // pour calculer et remplir la matrice deg
        calculerDeg(n, matrice);

        // traduire les arcs
        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++) {
                if (matrice[i][j] == 1) {
                    System.out.println("Node " + i + " is connected to " + j);
                }
            }
            System.out.println("--------------------------------------");
            System.out.println();
        }

    }
}
